From 689d182b8aec9279925842e78c1e71dcd16bbb31 Mon Sep 17 00:00:00 2001
From: Amir Vadai <avadai@redhat.com>
Date: Sun, 5 Oct 2014 17:04:46 -0400
Subject: [ethernet] mlx5: Add ConnectX-4 to list of supported devices

Message-id: <5ffadf63f4707de93b9beae9a9377b68e83d507c.1412528529.git.avadai@redhat.com>
Patchwork-id: 96906
O-Subject: [Patch RHEL7.1] net/mlx5_core: Add ConnectX-4 to list of supported devices
Bugzilla: 1149439
RH-Acked-by: Doug Ledford <dledford@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>
RH-Acked-by: David S. Miller <davem@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

From: Eli Cohen <eli@mellanox.com>

Add the upcoming ConnectX-4 device to the list of supported devices by then
mlx5 driver.

Signed-off-by: Eli Cohen <eli@mellanox.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit f832dc820fe8fca561933e8fa734adca75bba5a0)

Brew: 8063955
Bugzilla: 1149439
Tested: In Mellanox lab
Signed-off-by: Amir Vadai <avadai@redhat.com>
---
Although the hardware is not out yet, it will keep backward compatability to
Connect-IB, therefore current mlx5 driver just need to know the PCI id and
everything should work out of the box.

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/mellanox/mlx5/core/main.c b/drivers/net/ethernet/mellanox/mlx5/core/main.c
index f2716cc..98e874e 100644
--- a/drivers/net/ethernet/mellanox/mlx5/core/main.c
+++ b/drivers/net/ethernet/mellanox/mlx5/core/main.c
@@ -782,6 +782,7 @@ static void remove_one(struct pci_dev *pdev)
 
 static const struct pci_device_id mlx5_core_pci_table[] = {
  { PCI_VDEVICE(MELLANOX, 4113) }, /* MT4113 Connect-IB */
+ { PCI_VDEVICE(MELLANOX, 4115) }, /* ConnectX-4 */
  { 0, }
 };
 
-- 
1.7.1